### PR TITLE
[Core] Add Spans CurveOnSurface Intersection

### PR DESCRIPTION
**Pull request by tteschemacher**
Hi all,

I am adding here the functionalities to get the intersection of a curve with its underlying surface knots.

### CODE REVIEW

**Comment by philbucher** [comment]
No tests? ;) / :/

**Comment by tteschemacher** [comment]
There's a test :P

Which is even failing ;)

**Comment by oberbichler** [comment]
Was there a reason to use Bisection instead of Newton? When I wrote the algorithm, the NR was much more efficient.

I remember that there were many special cases to consider. The algorithm in ANurbs is already proven in many practical examples.

**Comment by philbucher** [comment]
> There's a test :P
> 
> Which is even failing ;)

Oh man I love the CI :)

**Comment by tteschemacher** [comment]
@oberbichler you're right the Newton raphson (which is even a reduced nr as it just goes against one coordinate) is more efficient. I was now keen on the interface. I need to have something running until Monday. I would be more than happy if you could put your implementation to it.

File: kratos/geometries/brep_curve_on_surface.h [hunk 1]
```diff
@@ -238,6 +238,21 @@ class BrepCurveOnSurface
...
+    ///@}
+    ///@name Curve Properties
+    ///@{
+
+    /* @brief Provides intersections of the nurbs curve with the knots of the surface,
+     *         using the interval of this curve.
+     * @param vector of span intervals.
+     * @param index of chosen direction, for curves always 1.
+     */
+    void Spans(std::vector<double>& rSpans, IndexType DirectionIndex = 1) const
```
Comment (pooyan-dadvand) [inline kratos/geometries/brep_curve_on_surface.h]: I know that for curve on surface you don't use it but for correct documentation the direction should be 0. Isn't it?

File: kratos/geometries/nurbs_surface_geometry.h [hunk 2]
```diff
@@ -254,6 +254,65 @@ class NurbsSurfaceGeometry : public Geometry<typename TContainerPointType::value
         return NumberOfKnotsV() - PolynomialDegreeV() + 1;
     }
 
+    /// Returns the number of spans in DirectionIndex=1:U and DirectionIndex=2:V (which are larger than 0).
```
Comment (pooyan-dadvand) [inline kratos/geometries/nurbs_surface_geometry.h]:
```suggestion
    /// Returns the number of spans in DirectionIndex=0:U and DirectionIndex=1:V (which are larger than 0).
```

**Review by pooyan-dadvand — COMMENTED** [review]
Just some cosmetic corrections

File: kratos/geometries/brep_curve_on_surface.h (hunk 1 above)
Comment (tteschemacher) [reply kratos/geometries/brep_curve_on_surface.h]: Yes, you're right!

**Comment by tteschemacher** [comment]
Thanks @pooyan-dadvand for your comments. I changed it accordingly.

**Review by pooyan-dadvand — APPROVED** [review]
👍

**Pull request merged**